### PR TITLE
Gpx issues #6222, #6140

### PR DESCRIPTION
**Pull request by MadWasp79**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/views/GPXLayer.java [hunk 1]
```diff
@@ -214,23 +217,25 @@ public void onPrepareBufferImage(Canvas canvas, RotatedTileBox tileBox, DrawSett
 	private int updatePaints(int color, boolean routePoints, boolean currentTrack, DrawSettings nightMode, RotatedTileBox tileBox) {
 		RenderingRulesStorage rrs = view.getApplication().getRendererRegistry().getCurrentSelectedRenderer();
 		final boolean isNight = nightMode != null && nightMode.isNightMode();
-		int hsh = calculateHash(rrs, routePoints, isNight, tileBox.getMapDensity(), tileBox.getZoom());
+		int hsh = calculateHash(rrs, routePoints, isNight, tileBox.getMapDensity(), tileBox.getZoom(),
+			view.getSettings().getCustomRenderProperty(CURRENT_TRACK_COLOR_ATTR).get(),
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/views/GPXLayer.java]: Extract to field of class

File: OsmAnd/src/net/osmand/plus/measurementtool/OptionsBottomSheetDialogFragment.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ public void onClick(View v) {
 
 		items.add(new DividerHalfItem(getContext()));
 
-		if (addLineMode) {
+		if (addLineMode && !snapToRoadEnabled) {
```
Comment (crimean) [inline OsmAnd/src/net/osmand/plus/measurementtool/OptionsBottomSheetDialogFragment.java]: We have Save button at top-right corner of the screen. Did you change it behaviour as well? Is it needed?

File: OsmAnd/src/net/osmand/plus/views/GPXLayer.java (hunk 1 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/views/GPXLayer.java]: fixed

File: OsmAnd/src/net/osmand/plus/measurementtool/OptionsBottomSheetDialogFragment.java (hunk 2 above)
Comment (MadWasp79) [reply OsmAnd/src/net/osmand/plus/measurementtool/OptionsBottomSheetDialogFragment.java]: changed behavior, now it opens "Save as new GPX file" dialog

**Pull request merged**